doc/guides/investigating_native_memory_leak.md

```diff
@@ -0,0 +1,434 @@
+# Investigating Memory Leaks with valgrind
+
+A Node.js process may run out of memory due to excessive consumption of
+native memory. Native Memory is memory which is not managed by the
+V8 Garbage collector and is allocated either by the Node.js runtime, its
+dependencies or native [addons](://nodejs.org/docs/latest/api/n-api.html).
```

Comment:
Is this link right? It's missing the protocol

Reply:
Thanks for catching that

Reply:
@XhmikosR fixed